go/vt/mysqlctl/backupengine.go

```diff
@@ -213,8 +219,14 @@ func registerBackupEngineFlags(fs *pflag.FlagSet) {
 // a particular backup by calling GetRestoreEngine().
 //
 // This must only be called after flags have been parsed.
-func GetBackupEngine() (BackupEngine, error) {
-	name := backupEngineImplementation
+func GetBackupEngine(backupEngine *string) (BackupEngine, error) {
+	var name string
+	if backupEngine == nil || *backupEngine == "" {
+		name = backupEngineImplementation
+	} else {
+		name = *backupEngine
+	}
+
```

Comment:
Let's make this more canonical (and also more idiomatic):
```suggestion
        name := backupEngineImplementation
	if backupEngine != "" {
		name = backupEngine
	}
```